packages/block-library/src/search/view.js

```diff
@@ -0,0 +1,65 @@
+window.addEventListener( 'DOMContentLoaded', () => {
+	const hiddenClass = 'wp-block-search__searchfield-hidden';
+
+	Array.from(
+		document.getElementsByClassName(
+			'wp-block-search__button-behavior-expand'
+		)
+	).forEach( ( block ) => {
+		const searchField = block.querySelector( '.wp-block-search__input' );
+		const searchButton = block.querySelector( '.wp-block-search__button' );
+		const activeElement = block.ownerDocument.activeElement;
+
+		const ariaLabel = searchButton.getAttribute( 'aria-label' );
+		const id = searchField.getAttribute( 'id' );
+
+		const toggleSearchField = ( showSearchField ) => {
+			if ( showSearchField ) {
+				searchField.removeAttribute( 'aria-hidden' );
+				searchField.removeAttribute( 'tabindex' );
+				searchButton.removeAttribute( 'aria-expanded' );
+				searchButton.removeAttribute( 'aria-controls' );
+				searchButton.removeAttribute( 'aria-label' );
+
+				return block.classList.remove( hiddenClass );
+			}
+
+			searchField.setAttribute( 'aria-hidden', 'true' );
+			searchField.setAttribute( 'tabindex', '-1' );
+			searchButton.setAttribute( 'aria-expanded', 'false' );
+			searchButton.setAttribute( 'aria-controls', id );
+			searchButton.setAttribute( 'aria-label', ariaLabel );
+			return block.classList.add( hiddenClass );
+		};
+
+		const hideSearchField = ( e ) => {
+			if (
+				! e.target.closest( '.wp-block-search__inside-wrapper' ) &&
+				activeElement !== searchButton &&
+				activeElement !== searchField
+			) {
+				return toggleSearchField( false );
+			}
+
+			if ( e.key === 'Escape' ) {
+				searchButton.focus();
+				return toggleSearchField( false );
+			}
+		};
+
+		const handleButtonClick = ( e ) => {
+			if ( block.classList.contains( hiddenClass ) ) {
+				e.preventDefault();
+				searchField.focus();
+				toggleSearchField( true );
+			}
+		};
+
+		searchField.addEventListener( 'blur', hideSearchField );
```

Comment:
This event listener doesn't seem to do anything, I think because of the way `hideSearchField` checks to see if `searchField` is the active element.

Reply:
You're right. I removed this event listener in [7665720](https://github.com/WordPress/gutenberg/pull/50487/commits/7665720906561a8b71ba59d3a468cd75fa9a413c) in since it feels jarring to hide the field solely on focus loss.